wodles/aws/tests/test_wazuh_integration.py

```diff
@@ -0,0 +1,462 @@
+
+    instance.check_metadata_version()
+    new_metadata_value = utils.database_execute_query(custom_database, instance.sql_get_metadata_version)
+    assert new_metadata_value == utils.WAZUH_VERSION
+
+
+def test_wazuh_aws_database_check_metadata_version_no_table(custom_database):
+    """Test if `check_metadata_version` function updates the metadata value when the table does not exist."""
+    instance = utils.get_mocked_wazuh_aws_database(db_name=utils.TEST_DATABASE)
+    instance.db_connector = custom_database
+    instance.db_cursor = instance.db_connector.cursor()
+    instance.check_metadata_version()
+    new_metadata_value = utils.database_execute_query(custom_database, instance.sql_get_metadata_version)
+    assert new_metadata_value == utils.WAZUH_VERSION
+
+
+@pytest.mark.parametrize('table_exists', [True, False, sqlite3.Error])
+def test_wazuh_aws_database_check_metadata_version_handles_exceptions(custom_database, table_exists):
+    """Test if `check_metadata_version` function handles exceptions properly.
+
+    Parameters
+    ----------
+    table_exists : bool or sqlite3.Error
+        The value to be returned by the mocked database call.
+    """
+    mocked_table_exists = MagicMock()
+    if isinstance(table_exists, bool):
+        mocked_table_exists.fetchone.return_value = table_exists
+    else:
+        mocked_table_exists.fetchone.side_effect = table_exists
+    mocked_cursor = MagicMock()
+    mocked_cursor.execute.side_effect = [mocked_table_exists, sqlite3.OperationalError]
+
+    instance = utils.get_mocked_wazuh_aws_database(db_name=utils.TEST_DATABASE)
+    instance.db_connector = custom_database
+    instance.db_cursor = mocked_cursor
+
+    with pytest.raises(SystemExit) as e:
+        instance.check_metadata_version()
+    assert e.value.code == utils.METADATA_ERROR_CODE
+
+
+def test_wazuh_aws_database_delete_deprecated_tables(custom_database):
+    """Test `delete_deprecated_tables` function remove unwanted tables while keeping the rest intact."""
+    # Populate the database
+    utils.database_execute_script(custom_database, TEST_METADATA_DEPRECATED_TABLES_SCHEMA)
+
+    instance = utils.get_mocked_wazuh_aws_database(db_name=utils.TEST_DATABASE)
+    instance.db_connector = custom_database
+    instance.db_cursor = instance.db_connector.cursor()
+
+    for table in wazuh_integration.DEPRECATED_TABLES:
+        assert instance.db_cursor.execute(instance.sql_find_table, {'name': table}).fetchone()[0]
+    assert instance.db_cursor.execute(instance.sql_find_table, {'name': METADATA_TABLE_NAME}).fetchone()[0]
+
+    instance.delete_deprecated_tables()
+
+    # The deprecated tables were deleted
+    for table in wazuh_integration.DEPRECATED_TABLES:
+        assert not instance.db_cursor.execute(instance.sql_find_table, {'name': table}).fetchone()
+    # The metadata table is still present
+    assert instance.db_cursor.execute(instance.sql_find_table, {'name': METADATA_TABLE_NAME}).fetchone()[0]
```

Comment:
Missing new line at the end of the file.
```suggestion
    assert instance.db_cursor.execute(instance.sql_find_table, {'name': METADATA_TABLE_NAME}).fetchone()[0]

```